File: staylio_frontend/common/interfaces/response/NotificationResponse.ts
import { NotificationType } from "@common/enums/NotificationType";

export interface NotificationResponse {
    id: number;
    senderId: number;
    senderName: string;
    senderAvatar: string;
    receiverId: number;
    title: string;
    content: string;
    type: NotificationType;
    referenceId?: number;
    isRead: boolean;
    readAt: string;
    createdAt: string;
}

File: staylio_backend/src/main/java/com/example/staylio_backend/dto/response/NotificationResponse.java
package com.example.staylio_backend.dto.response;

import com.example.staylio_backend.model.enums.NotificationType;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationResponse {

    private Long id;

    // SENDER
    private Long senderId;
    private String senderName;
    private String senderAvatar;

    // RECEIVER
    private Long receiverId;

    // CONTENT
    private String title;
    private String content;

    // TYPE
    private NotificationType type;

    // REFERENCE
    private Long referenceId;

    // STATUS
    private Boolean isRead;
    private LocalDateTime readAt;

    // TIME
    private LocalDateTime createdAt;
}